docs/installation/est/Installing_EST_pkispawn.md

```diff
@@ -0,0 +1,202 @@
+
+It is possible to use different schemas but in this case a custom
+`statements.conf` file (provided in the same folder) has to be
+provided in order to retrieve the user information from the DB.
+
+
+## Installation
+
+An example installation configuration is provided in
+`/usr/share/pki/server/examples/installation/est.cfg`. To install EST
+in the same instance of the CA and with the DS realm run the command:
+
+```
+$ pkispawn \
+    -f /usr/share/pki/server/examples/installation/est.cfg \
+    -s EST \
+    -D est_realm_url=ldap://estds.example.com:3389 \
+    -v
+```
+
+The `est_realm_url` points to the user DB. The other configurations that could be modified according to the deployment are:
+
+```
+est_ca_profile=estServiceCert
+est_ca_user_name=
+est_ca_user_password=
+est_ca_user_password_file=
+est_ca_user_certificate=
+est_realm_type=
+est_realm_custom=
+est_realm_url=
+est_realm_auth_type=BasicAuth
+est_realm_bind_dn=cn=Directory Manager
+est_realm_bind_password=
+est_realm_nickname=
+est_realm_user=
+est_realm_username=
+est_realm_password=
+est_realm_users_dn=ou=people,dc=est,dc=pki,dc=example,dc=com
+est_realm_groups_dn=ou=groups,dc=est,dc=pki,dc=example,dc=com
+est_realm_statements=/usr/share/pki/est/conf/realm/postgresql/statements.conf
+est_authorizer_exec_path=/usr/share/pki/est/bin/estauthz
+```
+
+The `est_ca_*` provides information related to the user and profile
+configured in the CA for the EST subsystem.
+
+The `est_authorizer_exec_path` is the executable responsible to verify
+the authorisation. The provided script checks only that the user has
+the role *estclient*.
+
+The `est_realm_*` options allow to customise the realm. Possible types
+are: ds, postgresql and in-memory. As an example, to install EST with
+PostgreSQL the command will be
+
+```
+pkispawn \
+    -f /usr/share/pki/server/examples/installation/est.cfg \
+    -s EST \
+    -D est_realm_url="jdbc:postgresql://postgresql.example.com:5432/est?ssl=true&sslmode=require" \
+    -D est_realm_type=postgresql \
+    -D est_realm_user=est \
+    -D est_realm_password=mysecretpassword \
+    -v
+```
+
+The `est_realm_custom` is a path to a custom realm configuration for
+tomcat and if provided it will overwrite all other realm related
+configurations.
+
+### Installation on separate instance
+
+When installed on a separate instance the CA URL and the signing
+certificate has to be provided. This is done with the command:
+
+```
+pkispawn \
+    -f /usr/share/pki/server/examples/installation/est.cfg \
+    -s EST \
+    -D est_realm_url=ldap://estds.example.com:3389 \
+    -D pki_ca_uri=https://ca.example.com:8443 \
+    -D pki_cert_chain_path=ca_signing.crt \
+    -D pki_cert_chain_nickname=caSigning \
+    -v
+```
+
+Note: the CA signing certificate can be obtained with the following command in the CA server:
+```
+pki-server cert-export ca_signing --cert-file ca_signing.crt
+```
+
+EST will get the SSL server certificate through the EST provided user
```

Comment:
I'm reading this comment a few times and have to make a guess.  Please verify my understanding. This paragraph is saying that without the two pkcs12 related parameters, pkispawn will get the EST's server cert automatically issued by using the estServerCert (the default name, which is hard-coded?) profile.  And if one wishes to use a pre-generated server cert/keys, then supply its p12 and passwd as stated in the pkcs12 params.

Reply:
@ladycfu your guess is correct. If it is not provided with a p12 then it will try to get a certificate using the EST credentials and profile which are not hard-coded but they are parameter (`est_ca_profile` and `est_ca_user_*`). Currently we do not support multiple profiles although the specification allows to include a label in a request which could be used to select different CAs (or profiles I think).

Reply:
@fmarco76 got it!  In that case, I suggest a little clarification.  Maybe something like "By default, the EST server cert will be issued based on the information provided via the pkispawn parameters est_ca_profile and est_ca_user_*.  Alternatively, the EST server cert could be pre-issued prior to pkispawn with its full chain in PKCS#12 bundle supplied via the pki_server_pkcs12_* parameters on the pkispawn command line as exemplified below. It is important tha tthe certificate aliases ..."